### PR TITLE
[10.0][IMP] event: Make event report inheritable.

### PR DESCRIPTION
**Pull request by sergio-teruel**
**Description of the issue/feature this PR addresses:**
This PR makes event report inheritable for other modules as sale report
**Current behavior before PR:**
extend event repor tis not possible
**Desired behavior after PR is merged:**
You can extend the event report

--
@mart-e Can you take a look?. Thanks
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
cc @Tecnativa

### CODE REVIEW

File: addons/event/report/report_event_registration.py [hunk 1]
```diff
@@ -27,34 +27,37 @@ class ReportEventRegistration(models.Model):
...
-                CASE WHEN r.state IN ('cancel') THEN count(r.event_id) ELSE 0 END AS cancel_state,
-                e.event_type_id AS event_type_id,
-                e.seats_max AS seats_max,
-                e.state AS event_state,
-                r.state AS registration_state
-            FROM
-                event_event e
+    def _from(self):
+        from_str = """
+            event_event e
```
Comment (pedrobaeza) [inline addons/event/report/report_event_registration.py]: Include here FROM word

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done.

File: addons/event/report/report_event_registration.py [hunk 2]
```diff
@@ -27,34 +27,37 @@ class ReportEventRegistration(models.Model):
...
-                e.event_type_id AS event_type_id,
-                e.seats_max AS seats_max,
-                e.state AS event_state,
-                r.state AS registration_state
+    def _from(self):
+        from_str = """
             FROM
-                event_event e
-                LEFT JOIN event_registration r ON (e.id=r.event_id)
+            event_event e LEFT JOIN event_registration r ON (e.id=r.event_id)
```
Comment (pedrobaeza) [inline addons/event/report/report_event_registration.py]: Why changing here the 2 lines?

File: addons/event/report/report_event_registration.py [hunk 3]
```diff
@@ -27,34 +27,37 @@ class ReportEventRegistration(models.Model):
     name_registration = fields.Char('Participant / Contact Name', readonly=True)
     company_id = fields.Many2one('res.company', 'Company', readonly=True)
 
-    @api.model_cr
-    def init(self):
-        """Initialize the sql view for the event registration """
-        tools.drop_view_if_exists(self.env.cr, 'report_event_registration')
+    # TOFIX this request won't select events that have no registration
+    def _select(self):
+        select_str = """SELECT
```
Comment (pedrobaeza) [inline addons/event/report/report_event_registration.py]: Don't change here the indentation

File: addons/event/report/report_event_registration.py (hunk 2 above)
Comment (sergio-teruel) [inline addons/event/report/report_event_registration.py]: Because it fit and it is short

Comment (pedrobaeza) [inline addons/event/report/report_event_registration.py]: Yeah, but making changes here the priority is to modify the minimum number of lines, and it doesn't comply with general rule applied in other reports (one line, one table)

**Comment by sergio-teruel** [comment]
@pedrobaeza changes done

**Comment by tivisse** [comment]
@sergio-teruel Thanks for the feedback.

By the way I don't understand why you make a query and a subquery for the SELECT. Furthermore, I won't do it in 10.0 but in master instead.

Waiting your input.

**Comment by sergio-teruel** [comment]
@tivisse Thanks for your review, In a first commit I did not use a subquery, but in other modules that depend of event I had needed make calculation over the fields, also by this way i can add other models to query and make compute fields in the query, i think this way is more cleaner.

**Comment by tivisse** [comment]
Well I still don't understand why you make a second request to retrieve all the elements you selected in the sub request. It's not correct performance wise. 

IMHO you should have only one select.

**Comment by pedrobaeza** [comment]
@tivisse are there any chance that this can land on 10.0 if we fix it? Should we pass an OPW ticket?

**Comment by tivisse** [comment]
@pedrobaeza If we remove the first query, I have no objections to merge this in 10.0 :)

**Comment by pedrobaeza** [comment]
Hi @tivisse, I have fixed the PR, removing the subquery. I have also checked that the `FIXME` comment is no longer applied, as events without registrations also appear on the query.

**Comment by tivisse** [comment]
@sergio-teruel @pedrobaeza 

As the report doesn't exist anymore in master (I suppose to use a pivot view instead), I merged (exceptionally) your PR on 10.0 at https://github.com/odoo/odoo/commit/18152f35b6a0db7d710c41c7a1040808d7e36bde

Have a nice day.

**Comment by pedrobaeza** [comment]
Thank you very much!

I see the new approach of using pivot view over event.event with computed fields for the quantities. Indeed it seems better.

**Pull request closed without merging**